pkg/syncer/util.go

```diff
@@ -364,3 +385,42 @@ func SyncerInitConfigInfo(ctx context.Context) (*cnsconfig.ConfigurationInfo, er
 	}
 	return configInfo, nil
 }
+
+// getVcHostAndVolumeManager returns VC host and the corresponding volume manager that can access the given volume on VC.
+// In case of a single VC setup, we simply return the metadataSyncer's host and volumeManager fields.
+func getVcHostAndVolumeManager(ctx context.Context, metadataSyncer *metadataSyncInformer, volumeID string) (string, volumes.Manager, error) {
+	log := logger.GetLogger(ctx)
+
+	// isMultiVCenterFssEnabled feature gate is always going to be disabled for flavors other than vanilla.
+	if !isMultiVCenterFssEnabled {
+		return metadataSyncer.host, metadataSyncer.volumeManager, nil
+	}
+
+	if len(metadataSyncer.configInfo.Cfg.VirtualCenter) == 1 {
+		vCenter := metadataSyncer.configInfo.Cfg.Global.VCenterIP
+		cnsVolumeMgr, volMgrFound := metadataSyncer.volumeManagers[vCenter]
+		if !volMgrFound {
+			return "", nil, logger.LogNewErrorf(log,
+				"could not get volume manager for the vCenter: %q", vCenter)		
+		}
+		return vCenter, cnsVolumeMgr, nil
+	}
+
+	if volumeInfoService != nil {
+		vCenter, err := volumeInfoService.GetvCenterForVolumeID(ctx, volumeID)
+		if err != nil {
+			log.Errorf("failed to get vCenter for the volumeID: %q with err=%+v", volumeID, err)
+			return "", nil, logger.LogNewErrorf(log,
+				"failed to get vCenter for the volumeID: %q with err=%+v", volumeID, err)
+		}
+		volumeManager, volumeManagerfound := metadataSyncer.volumeManagers[vCenter]
+		if !volumeManagerfound {
+			return "", nil, logger.LogNewErrorf(log,
+				"could not get volume manager for the vCenter: %q", vCenter)
+		}
+		return vCenter, volumeManager, nil
+	}
+
+	log.Errorf("failed to get VC host and volume manager. Possibly, volumeInfoService is nil.")
```

Comment:
"Possibly, volumeInfoService is nil" => It is confirmed that volumeInfoService is nil here.

Reply:
We do not need to print error here, as we are already using logger.LogNewErrorf in the next line.